src/BenchmarkDotNet/Environments/Runtimes/MonoAotLLVMRuntime.cs

```diff
@@ -0,0 +1,46 @@
+﻿using System;
+using System.ComponentModel;
+using System.IO;
+using BenchmarkDotNet.Extensions;
+using BenchmarkDotNet.Jobs;
+
+namespace BenchmarkDotNet.Environments
+{
+    public class MonoAotLLVMRuntime : Runtime, IEquatable<MonoAotLLVMRuntime>
+    {
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        internal static readonly MonoAotLLVMRuntime Default = new MonoAotLLVMRuntime();
+
+        public FileInfo AOTCompilerPath { get; }
+
+        /// <summary>
+        /// creates new instance of MonoAotLLVMRuntime
+        /// </summary>
+        /// <param name="mainJs">MANDATORY path to the main.js file.</param>
+        /// <remarks>path to mainJs MUST be provided</remarks>
+        public MonoAotLLVMRuntime(FileInfo aotCompilerPath, string msBuildMoniker = "net6.0", string displayName = "MonoAOTLLVM") : base(RuntimeMoniker.MonoAOTLLVM, msBuildMoniker, displayName)
+        {
+            if (aotCompilerPath == null)
+                throw new ArgumentNullException(paramName: nameof(aotCompilerPath));
+            if (aotCompilerPath.IsNotNullButDoesNotExist())
+                throw new FileNotFoundException($"Provided {nameof(aotCompilerPath)} file: \"{aotCompilerPath.FullName}\" doest NOT exist");
+
+            AOTCompilerPath = aotCompilerPath;
+        }
+
+        // this ctor exists only for the purpose of having .Default property that returns something consumable by RuntimeInformation.GetCurrentRuntime()
```

Comment:
You mention `RuntimeInformation.GetCurrentRuntime`, but seem to forget to include the changes in this method in this PR.

Reply:
Actually I think I cribbed this comment from wasm; do we actually need any change to RuntimeInformation?

Reply:
>  do we actually need any change to RuntimeInformation

Yes, otherwise BDN will print invalid data in the table (it will recognize the runtime as .NET 6?)

Reply:
So, to do this, I'd need to have some way for C# code to know if it is being run in AOT? I think right now it actually hits the mono case (but says nothing about AOT).

Reply:
Hmm, I can't actually find a way to detect if we are using AOT here. I could use `System.Runtime.CompilerServices.RuntimeFeature`, but its not available in netstandard2.0.

It seems like it will detect 'mono', which is the runtime, but not the fact that it is AOT.

Reply:
> I could use System.Runtime.CompilerServices.RuntimeFeature, but its not available in netstandard2.0.

In such a case you can use reflection to read the data:

```cs
private static bool IsAOT()
{
    Type runtimeFeature = Type.GetType("System.Runtime.CompilerServices.RuntimeFeature");
    if (runtimeFeature != null)
    {
        MethodInfo methodInfo = runtimeFeature.GetProperty("IsDynamicCodeCompiled", BindingFlags.Public | BindingFlags.Static)?.GetMethod;
        if (methodInfo != null)
        {
            return (bool)methodInfo.Invoke(null, null);
        }
    }

    return false;
}
```